config/types.go

```diff
@@ -114,6 +114,111 @@ type Config struct {
+		KeyFile                  string  `yaml:"key_file"                      env:"ETCD_KEY_FILE"`
+		ListenClientURLs         string  `yaml:"listen_client_urls"            env:"ETCD_LISTEN_CLIENT_URLS"          deprecated:"etcd2 options no longer work for etcd"`
+		ListenPeerURLs           string  `yaml:"listen_peer_urls"              env:"ETCD_LISTEN_PEER_URLS"            deprecated:"etcd2 options no longer work for etcd"`
+		MaxResultBuffer          int     `yaml:"max_result_buffer"             env:"ETCD_MAX_RESULT_BUFFER"`
+		MaxRetryAttempts         int     `yaml:"max_retry_attempts"            env:"ETCD_MAX_RETRY_ATTEMPTS"`
+		MaxSnapshots             int     `yaml:"max_snapshots"                 env:"ETCD_MAX_SNAPSHOTS"               deprecated:"etcd2 options no longer work for etcd"`
+		MaxWALs                  int     `yaml:"max_wals"                      env:"ETCD_MAX_WALS"                    deprecated:"etcd2 options no longer work for etcd"`
+		Name                     string  `yaml:"name"                          env:"ETCD_NAME"`
+		PeerAddr                 string  `yaml:"peer_addr"                     env:"ETCD_PEER_ADDR"`
+		PeerBindAddr             string  `yaml:"peer_bind_addr"                env:"ETCD_PEER_BIND_ADDR"`
+		PeerCAFile               string  `yaml:"peer_ca_file"                  env:"ETCD_PEER_CA_FILE"`
+		PeerCertFile             string  `yaml:"peer_cert_file"                env:"ETCD_PEER_CERT_FILE"`
+		PeerElectionTimeout      int     `yaml:"peer_election_timeout"         env:"ETCD_PEER_ELECTION_TIMEOUT"`
+		PeerHeartbeatInterval    int     `yaml:"peer_heartbeat_interval"       env:"ETCD_PEER_HEARTBEAT_INTERVAL"`
+		PeerKeyFile              string  `yaml:"peer_key_file"                 env:"ETCD_PEER_KEY_FILE"`
+		Peers                    string  `yaml:"peers"                         env:"ETCD_PEERS"`
+		PeersFile                string  `yaml:"peers_file"                    env:"ETCD_PEERS_FILE"`
+		Proxy                    string  `yaml:"proxy"                         env:"ETCD_PROXY"                       deprecated:"etcd2 options no longer work for etcd"`
+		RetryInterval            float64 `yaml:"retry_interval"                env:"ETCD_RETRY_INTERVAL"`
+		Snapshot                 bool    `yaml:"snapshot"                      env:"ETCD_SNAPSHOT"`
+		SnapshotCount            int     `yaml:"snapshot_count"                env:"ETCD_SNAPSHOTCOUNT"`
+		StrTrace                 string  `yaml:"trace"                         env:"ETCD_TRACE"`
+		Verbose                  bool    `yaml:"verbose"                       env:"ETCD_VERBOSE"`
+		VeryVerbose              bool    `yaml:"very_verbose"                  env:"ETCD_VERY_VERBOSE"`
+		VeryVeryVerbose          bool    `yaml:"very_very_verbose"             env:"ETCD_VERY_VERY_VERBOSE"`
+	} `yaml:"etcd"`
+	Etcd2 struct {
```

Comment:
That seems reasonable. I wonder if there is a way we can easily discover that information from etcd itself, instead of having to maintain both etcd and these PV pairs. /cc @heyitsanthony